src/util/mod.rs

```diff
@@ -46,6 +47,34 @@ impl DataWriter {
     }
 }
 
+pub struct BodyWriter {
+    directory: String
+}
+
+impl BodyWriter {
+    /// creates a new directory in the current working path
+    pub fn new(directory: &str) -> BodyWriter {
+        if !Path::new(directory).exists() {
+            fs::create_dir(directory).expect("Could'nt create dir.");
+        }
+
+        BodyWriter {
+            directory: directory.to_owned()
+        }
+    }
+
+    pub fn write_bodies(&self, bodies: Vec<Body> , path: String) -> std::io::Result<()> {
```

Comment:
The argument `path` is a bit misleading. It's being used as a file name, so it would be better to call it `filename` or something like that.